checkov/common/checks_infra/solvers/complex_solvers/base_complex_solver.py

```diff
@@ -25,13 +25,20 @@ def _get_operation(self, *args: Any, **kwargs: Any) -> Any:
     def _get_negative_op(self, *args: Any) -> Any:
         return not self._get_operation(args)
 
-    def run(self, graph_connector: DiGraph) -> Tuple[List[Dict[str, Any]], List[Dict[str, Any]]]:
+    def get_operation(self, vertex: Dict[str, Any]) -> Optional[bool]:
```

Comment:
please make the method abstract
```suggestion
    @abstractmethod
    def get_operation(self, vertex: Dict[str, Any]) -> Optional[bool]:
```